utils/utils.go

```diff
@@ -133,9 +134,16 @@ func FromMapValueToStringMap(tl types.Map) map[string]string {
 }
 
 func Int64To32Pointer(in types.Int64) *int32 {
+	n32 := Int64To32(in)
+	return &n32
+}
+
+func Int64To32(in types.Int64) int32 {
 	n64 := in.ValueInt64()
-	var n2 = int32(n64)
-	return &n2
+	if n64 >= math.MinInt32 && n64 <= math.MaxInt32 {
+		return int32(n64)
+	}
+	panic(fmt.Sprintf("int64 value %d is out of range for int32", n64))
```

Comment:
As a general rule, I just like to point out that panics are rarely the best outcome. Most of the times we can do with a (int32, error) return types since that does not crash the system

Reply:
This is intentional, I don't want to pass back and handle errors for an issue that is extremely unlikely to happen under normal circumstances. Altough panics are not great, it is better to do something in the rare event of a conversion issue. Ofc in terms of node count and similar the code most probably won't panic, no one needs to set higher than 2147483647 values in these modules.

Reply:
However you are right, it is not a good practice to panic, if we want to handle an error, we should pass it back. This is just a special case where I believe it makes sense to keep things simple.